cmd/project.go

```diff
@@ -105,6 +108,38 @@ var projectCreateCmd = &cobra.Command{
 	},
 }
 
+var projectDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete a project",
+	Long:  "Delete a project and all resources deployed in the project being deleted",
+	Example: `  # Delete a project
+  odo project delete myproject
+	`,
+	Args: cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		projectName := args[0]
```

Comment:
we can check here whether given project name exist or not using `project.Exists` ?

Reply:
if don't exist, don't proceed :smile:

Reply:
it's better to check at early stage ?

Reply:
Not sure how validation in pkg/project is not early stage
I would say any call or invocation out of odo for ex: api calls, cmd invocation may be using oc(I agree we have moved away from this but considering oc just for the purpose of an example) would be the border to consider anything as early or delayed
But I might be missing something. So, fixed it as per the comment..

Reply:
@surajnarwade is right, similar to other checks such as in `create.go` here: https://github.com/redhat-developer/odo/blob/master/cmd/create.go#L92 we should have the Exists code appear in `cmd`.

Reply:
Nevermind, I see you already added this in line 124